File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var goodNodes = function(root) {
    let ans = 0;
    const dfs = (now, maxNum) => {
        if (now.val >= maxNum){
            ans += 1;
        }
        if (now.left){
            dfs(now.left, Math.max(maxNum, now.val));
        }
        if (now.right){
            dfs(now.right, Math.max(maxNum, now.val));
        }
    }
    dfs(root, -10000);
    return ans;
};

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.path = []
        self.goodNodesCount = 0
    
    def goodNodes(self, root: TreeNode) -> int:
        _max = max(self.path) if self.path else -10001
        if _max <= root.val: self.goodNodesCount += 1
        self.path.append(root.val)
        if root.left:
            self.goodNodes(root.left)
            self.path.pop()
        if root.right:
            self.goodNodes(root.right)
            self.path.pop()
        return self.goodNodesCount
        
        
        